dispatcher/dispatcher.go

```diff
@@ -155,8 +178,11 @@ Loop:
 			go func(job jobqueue.Job) {
 				defer wg.Done()
 				defer func() { <-d.sem }()
-				rslt := d.worker.Work(job)
-				d.jobqueue.Complete(job, rslt)
+				err := d.limiter.Wait(ctx)
+				if err == nil {
```

Comment:
[QUESTION]

It looks that the `job` will be lost when it exceeded the burst size.  Shouldn't we retry the `job` later?

Reply:
No, I think we shoudn't.

https://pkg.go.dev/golang.org/x/time/rate#Limiter.WaitN

> It returns an error if n exceeds the Limiter's burst size

In this case, n is always 1 because we use `Wait`. Although we have to care for burst size = 0, this is a broken job queue basically (0 burst size allows no execution). We will never encounter the case.

However, as you worry about, some jobs would stay `grabbed` when the dispatcher is stopped. Should we do something? If so, is there a recommended way to recover?

Reply:
> We will never encounter the case.

Ah, I see, I've misunderstood about this.

> If so, is there a recommended way to recover?

We don't have to worry about this.  We already have [a mechanism to recover `grabbed` jobs in primary-backup job queue activation](https://github.com/fireworq/fireworq/blob/de5b2e406499b3684c330b90b513302e301f2a6d/jobqueue/mysql/primary_backup.go#L30).